pandas/core/computation/eval.py

```diff
@@ -232,8 +237,23 @@ def eval(expr, parser='pandas', engine=None, truediv=True,
     pandas.DataFrame.query
     pandas.DataFrame.eval
     """
-    inplace = validate_bool_kwarg(inplace, 'inplace')
-    first_expr = True
+
+    def is_modifiable(obj):
+        target_type = type(obj).__name__
+        return hasattr(obj, "__dict__") or target_type == "dict"
+
+    inplace = validate_bool_kwarg(inplace, "inplace")
+
+    # "Primitives" cannot be modified inplace because
+    # they have no attributes that we can modify.
+    #
+    # The only exceptions are lists and dictionaries,
```

Comment:
Once I can rewrite the implementation in a way that works, I will add the tests (I have them in a separate file that I'm running to test before pushing).